### PR TITLE
Exif section corrections

### PR DESCRIPTION
**Pull request by artemShelest**
Sometimes PEL fails to determine APP1 section that contains EXIF data from the malformed JPEG. For instance. when APP1 section that contains XMP data precedes EXIF APP1 section.

### CODE REVIEW

**Comment by weberhofer** [comment]
An test case would be good, too.

File: src/PelJpeg.php [hunk 1]
```diff
@@ -401,16 +404,18 @@ public function getICC()
...
-        for ($i = 0; $i < $sections_count; $i ++) {
-            if ($this->sections[$i][0] == PelJpegMarker::APP1) {
-                unset($this->sections[$i]);
-                return;
+        $idx = 0;
+        while ($idx < count($this->sections)) {
+            $s = $this->sections[$idx];
+            if (($s[0] == PelJpegMarker::APP1) && ($s[1] instanceof PelExif)) {
+                array_splice($this->sections, $idx, 1);
+                $idx--;
```
Comment (weberhofer) [inline src/PelJpeg.php]: Why is $idx decremented here?

File: src/PelJpeg.php [hunk 2]
```diff
@@ -374,9 +374,12 @@ public function setICC(PelJpegContent $icc)
...
     {
-        $exif = $this->getSection(PelJpegMarker::APP1);
-        if ($exif instanceof PelExif) {
-            return $exif;
+        $sections_count = count($this->sections);
+        for ($i = 0; $i < $sections_count; $i ++) {
+            $section = $this->getSection(PelJpegMarker::APP1, $i);
+            if ($section instanceof PelExif) {
+                return $section;
+            }
```
Comment (weberhofer) [inline src/PelJpeg.php]: This function now returns the first valid Exif section? Please add a note to the method's header.

Comment (artemShelest) [reply src/PelJpeg.php]: Done

File: src/PelJpeg.php (hunk 1 above)
Comment (artemShelest) [reply src/PelJpeg.php]: Index has to be rewind after an item was removed, as following items' are shifted back. Reflected this in the test.

**Comment by artemShelest** [comment]
Fixed setExif() method as well.

File: src/PelJpeg.php (hunk 2 above)
Comment (weberhofer) [reply src/PelJpeg.php]: thanks

File: src/PelJpeg.php (hunk 1 above)
Comment (weberhofer) [reply src/PelJpeg.php]: ok

**Comment by weberhofer** [comment]
Thank you

**Pull request merged**